components/uniqueness/uniqueness-checker-client-service-impl/src/main/kotlin/net/corda/uniqueness/client/impl/LedgerUniquenessCheckerClientServiceImpl.kt

```diff
@@ -140,21 +140,23 @@ class LedgerUniquenessCheckerClientServiceImpl @Activate constructor(
     }
 
     // TODO CORE-9469 The key selection here will be replaced with the Crypto API once it is finished.
-    private fun selectNotaryVNodeSigningKey(notaryServiceKeys: List<PublicKey>): PublicKey {
-        val myInfo = memberLookup.myInfo()
+    private fun selectNotaryVNodeSigningKey(notaryServiceKey: PublicKey): PublicKey {
+        val signingKeys = signingService.findMySigningKeys(setOf(notaryServiceKey))
 
-        require(myInfo.notaryKeys.isNotEmpty()) {
-            "Could not find notary keys to sign with."
+        // We need to make sure there's at least one key we can sign with
+        require(signingKeys.any { it.value != null }) {
+            "Could not find any keys that belongs to the notary service's public key."
         }
 
-        // Always use the latest registered notary key. Even if key was rotated, the first element will be the latest.
-        val signingNotaryKey = myInfo.notaryKeys.first()
+        // We select the first key that is not null
+        val selectedSigningKey = signingKeys.values.filterNotNull().first()
```

Comment:
Lines 147 and 151 appear to be doing repeated work. It might be cleaner to condense all of this into something like:

```suggestion
        val selectedSigningKey = signingService
            .findMySigningKeys(setOf(notaryServiceKey)).values
            .filterNotNull()
            .firstOrNull()
            
        require(selectedSigningKey != null) {
            ...
        }
```

Reply:
Good point!